### PR TITLE
Gradient Angle

### PR DESCRIPTION
**Pull request by KyroVibe**
### Description
Added gradient angle and "spread" factor for more customization over gradients.

### CODE REVIEW

File: engine/Assets/Materials/Shaders/ImageGradient/ImageGradient.shader [hunk 1]
```diff
@@ -52,24 +55,32 @@ Shader "ImageGradient/ImageGradient" {
...
+
+                float4 gradientDirection = float4(cos(_GradientAngle), sin(_GradientAngle), 0, 1);
+                gradientDirection *= _GradientSpread;
+
+                float2 uvOffset = (gradientDirection.xy / 2) - float2(0.5 + _Offset.x, 0.5 + _Offset.y);
+
+                float4 col = lerp(_StartColor, _EndColor, clamp(dot(gradientDirection.xy, i.uv + uvOffset), 0, 1));
+                // float4 col = lerp(_StartColor, _EndColor, _Horizontal > 0 ? i.uv.x : 1 - i.uv.y);
+
+                //return float4(col.x, col.y, col.z, alpha*maskAlpha);
```
Comment (BrandonPacewic) [inline engine/Assets/Materials/Shaders/ImageGradient/ImageGradient.shader:82]:
```suggestion
```

**Comment by KyroVibe** [comment]
@BrandonPacewic The Clang CI is failing for a new reason

**Review by LucaHaverty — APPROVED** [review]
Everything seems to work well, nice job!

**Review by a-crowell — APPROVED** [review]

**Review by BrandonPacewic — APPROVED** [review]
Don't worry about the formatting action. It will be fixed in #904. Just ensure that the auto formatter is run before you merge.

**Pull request merged**